components/web/services/org.wso2.carbon.apimgt.authenticator/src/main/java/org/wso2/carbon/apimgt/authenticator/AuthenticatorAPI.java

```diff
@@ -85,15 +90,18 @@ public Response authenticate(
             NewCookie cookie2 = new NewCookie(AuthenticatorConstants.TOKEN_2,
                     part2 + "; path=" + restAPIContext + "; " + AuthenticatorConstants.HTTP_ONLY_COOKIE + "; "
                             + AuthenticatorConstants.SECURE_COOKIE);
-            NewCookie refreshTokenCookie = null;
+            NewCookie refreshTokenCookie1, refreshTokenCookie2 = null;
             if (refreshToken != null) {
-                refreshTokenCookie = new NewCookie(AuthenticatorConstants.REFRESH_TOKEN,
-                        refreshToken + "; path=" + appContext + "; " + AuthenticatorConstants.HTTP_ONLY_COOKIE + "; "
+                String refTokenPart1 = refreshToken.substring(0, refreshToken.length() / 2);
+                String refTokenPart2 = refreshToken.substring(refreshToken.length() / 2);
+                refreshTokenCookie1 = new NewCookie(AuthenticatorConstants.REFRESH_TOKEN_1,
+                        refTokenPart1 + "; path=" + appContext + "; " + AuthenticatorConstants.SECURE_COOKIE);
```

Comment:
Shall we use Sting builder for this since this has six String Concatenations in this 

refTokenPart2 + "; path=" + appContext + "; " + AuthenticatorConstants.HTTP_ONLY_COOKIE + "; "
As an example 
String a = b + c + d;
     into 
String a = new StringBuilder(b).append(c).append(d).toString();